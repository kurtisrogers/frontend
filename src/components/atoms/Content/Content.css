.content-grid,
.content-grid.full {
  display: grid;
  grid-template-columns: 
    [full-width-start] 100px [breakout-start] 100px [content-start] 1fr [content-end] 100px [breakout-end] 100px [full-width-end];
}

.content-grid > *,
.content-grid.full {
  grid-column: content;
}

.content-grid.wide > * {
  grid-column: breakout;
}

.content-grid.full > * {
  grid-column: full-width;
}