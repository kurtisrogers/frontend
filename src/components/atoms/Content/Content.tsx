import { handleColourClasses } from "@/helpers/colours";
import type { Branding } from "@/types/branding";

interface Props {
  content: any;
  children: any;
  gridLayout?: "wide" | "full";
  backgroundColor?: Branding["colors"];
}

export default function Content( props: Readonly<Props> ) {
  const { content, children, gridLayout, backgroundColor } = props;

  const colourClasses = handleColourClasses(backgroundColor, 'background');

  return (
    <section class={`${gridLayout ?? ''} content-grid ${colourClasses}`}>
      {content}
      {children}
    </section>
  )
}
