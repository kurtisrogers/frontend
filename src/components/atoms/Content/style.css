.content-grid > *,
.content-grid.full {
  grid-column: content;
  word-wrap: break-word;
  width: 100%;
}

.content-grid.wide > * {
  grid-column: breakout;
}

.content-grid:not(.banner) {
  display: flex;
  padding: 1rem;
}

.content-grid.full > * {
  grid-column: full-width;
}

.content-grid--light {
  color: var(--brand-black);
}

.content-grid--dark {
  color: var(--brand-white);
}

@media screen and (min-width: 1024px) {
  .content-grid,
  .content-grid:not(.banner),
  .content-grid.full {
    display: grid;
    grid-template-columns: 
      [full-width-start] 100px [breakout-start] 100px [content-start] 1fr [content-end] 100px [breakout-end] 100px [full-width-end];
  }
}
