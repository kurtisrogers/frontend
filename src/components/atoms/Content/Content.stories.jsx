import Content from "./index";
import "./style.css";

export default {
  title: "Atoms/Content",
  component: Content,
  argTypes: {
    variant: {
      options: ["yellow", "white", "grey", "black", "red", "default", undefined],
      control: { type: "select" }
    },
    gridLayout: {
      options: ["grid", "content", undefined],
      control: { type: "select" }
    }
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/solid/writing-docs/docs-page
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/solid/configure/story-layout
    layout: "fullscreen"
  }
};

const Template = (args) => (
  <Content variant={args.variant} gridLayout={args.gridLayout}>
    {args.children}
  </Content>
);

export const Default = Template.bind({});
Default.args = {
  variant: "red",
  gridLayout: "content",
  children: <p>Hello, I'm some content!</p>
};
