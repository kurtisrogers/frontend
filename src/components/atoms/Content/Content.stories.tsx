import Content from "./index";
import "./style.css";

export default {
  title: "Atoms/Content",
  component: Content,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/solid/writing-docs/docs-page
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/solid/configure/story-layout
    layout: "fullscreen"
  }
};

const Template = () => {
  return (
    <>
      <Content backgroundColor="yellow">
        <p>Hello, I'm some content!</p>
      </Content>
      <Content backgroundColor="white">
        <p>Hello, I'm some content!</p>
      </Content>
      <Content backgroundColor="grey">
        <p>Hello, I'm some content!</p>
      </Content>
      <Content backgroundColor="black">
        <p>Hello, I'm some content!</p>
      </Content>
      <Content backgroundColor="red">
        <p>Hello, I'm some content!</p>
      </Content>
      <Content>
        <p>Hello, I'm some content!</p>
      </Content>
    </>
  );
};

export const Default = Template.bind({});
// Default.args = {
//   // Add any props you want to pass to the Header component
// };
