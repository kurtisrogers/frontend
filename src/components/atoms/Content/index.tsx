import { handleColourClasses } from "@/helpers/colours";
import type { Branding } from "@/types/branding";
import type { gridLayoutOptions } from "@/types/grid";
import { JSX } from "solid-js";
import "./style.css";

export interface Props {
  children: JSX.Element;
  gridLayout?: gridLayoutOptions;
  variant?: Branding["colors"];
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}

export default function Content(props: Readonly<Props>) {
  const { children, gridLayout, variant } = props;

  const colourClasses = handleColourClasses(variant, "background");

  return <section class={`${gridLayout ?? ""} content-grid ${colourClasses}`}>{children}</section>;
}
