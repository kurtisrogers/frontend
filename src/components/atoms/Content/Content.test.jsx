// src/components/LinkButton.test.jsx
import { render } from '@solidjs/testing-library';
import Content from '.';
import { describe, it, expect } from 'vitest';

const props = {
  component: Content,
  children: (
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus quis sodales arcu. Ut nec
      imperdiet orci. Mauris ullamcorper lacus tincidunt dolor semper interdum. Nulla luctus ex
      vel volutpat ultricies. Mauris vel magna diam. Quisque non congue diam. Sed neque lacus,
      auctor vitae arcu eu, pellentesque condimentum lorem. Donec at rutrum mauris. Fusce
      molestie, metus vel aliquet elementum, felis dolor placerat nisi, a ultricies eros ipsum
      id lacus. Donec eget dolor lacus.
    </p>
  ),
  gridLayout: "wide",
  variant: "white"
}

describe('<Content />', () => {
  it('renders some simple content', () => {
    const { getByText } = render(() => <Content {...props} />);
    const content = getByText(/Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus quis sodales arcu. Ut nec imperdiet orci. Mauris ullamcorper lacus tincidunt dolor semper interdum. Nulla luctus ex vel volutpat ultricies. Mauris vel magna diam. Quisque non congue diam. Sed neque lacus, auctor vitae arcu eu, pellentesque condimentum lorem. Donec at rutrum mauris. Fusce molestie, metus vel aliquet elementum, felis dolor placerat nisi, a ultricies eros ipsum id lacus. Donec eget dolor lacus./);
    expect(content).toMatchSnapshot();
    expect(content).toBeInTheDocument();
  });
});
