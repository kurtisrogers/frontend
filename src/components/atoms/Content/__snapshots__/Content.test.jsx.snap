// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Content /> > renders some simple content 1`] = `
<p>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus quis sodales arcu. Ut nec imperdiet orci. Mauris ullamcorper lacus tincidunt dolor semper interdum. Nulla luctus ex vel volutpat ultricies. Mauris vel magna diam. Quisque non congue diam. Sed neque lacus, auctor vitae arcu eu, pellentesque condimentum lorem. Donec at rutrum mauris. Fusce molestie, metus vel aliquet elementum, felis dolor placerat nisi, a ultricies eros ipsum id lacus. Donec eget dolor lacus.
</p>
`;
